lib/matplotlib/tests/test_widgets.py

```diff
@@ -1464,6 +1478,45 @@ def onselect(vertices):
         do_event(tool_ref, etype, **event_args)
 
 
+def test_polygon_selector_box():
+    # Create a diamond shape
+    verts = [(20, 0), (0, 20), (20, 40), (40, 20)]
+    event_sequence = (polygon_place_vertex(*verts[0]) +
+                      polygon_place_vertex(*verts[1]) +
+                      polygon_place_vertex(*verts[2]) +
+                      polygon_place_vertex(*verts[3]) +
+                      polygon_place_vertex(*verts[0]))
+
+    ax = get_ax()
+
+    def onselect(vertices):
+        pass
+
+    # Create selector
+    tool = widgets.PolygonSelector(ax, onselect, draw_box=True)
+    for (etype, event_args) in event_sequence:
+        do_event(tool, etype, **event_args)
+
+    # Scale using the top right corner of the bounding box
+    do_event(tool._box, 'press', xdata=40, ydata=40, button=1)
+    do_event(tool._box, 'onmove', xdata=20, ydata=20, button=1)
+    do_event(tool._box, 'release', xdata=20, ydata=20, button=1)
+    assert tool.verts == [(10, 0), (0, 10), (10, 20), (20, 10)]
```

Comment:
This is failing because the `motion_notify_event` is not triggered, I think?

Reply:
I think that's right. I'm struggling to check how I can get it to trigger though. Because `do_event` goes directly to the widget `onmove`, it doesn't trigger any canvas events. I've tried
```python
    mock_event(ax, button='down', xdata=40, ydata=40, step=0)
    mock_event(ax, button=1, xdata=20, ydata=20, step=0)
    mock_event(ax, button='up', xdata=20, ydata=20, step=0)
```
instead, but that doesn't seem to be working either (the box itself doesn't even resize with this).

Reply:
I don't know if this is the right way to doing it but adding `ax.figure.canvas.motion_notify_event(0, 0)` is working.

Reply:
This and associated methods works, thanks for the tip.